cpp/pybind/CMakeLists.txt

```diff
@@ -1,31 +0,0 @@
-set(PYBIND11_NEWPYTHON ON)
```

Comment:
You were using FindPython before, too, that's that this setting does.

Reply:
Yes, but the ones I know of should only affect PyPy, and only for stand-alone builds (scikit-build-core makes sure FindPython works on PyPy). Do you have logs for a pybind11 build? What might be happening is that the library was never found, but nanobind needs it because it's generating a list of symbols to ignore, while pybind11 just ignores all unresolved symbols.

Reply:
Log here: https://github.com/mlc-ai/xgrammar/actions/runs/13894029090/job/38870730963- the raw log is [here](https://productionresultssa0.blob.core.windows.net/actions-results/86186adc-905f-4a32-9927-0ce36a5ee561/workflow-job-run-f163f28b-5b5d-512e-953a-98b8f90fe4e8/logs/job/job-logs.txt?rsct=text%2Fplain&se=2025-03-17T14%3A58%3A48Z&sig=8uPfYZmCWMDJ9u%2F%2Bk4NYecDn%2BYOZ6xGnpC3kteyg79c%3D&ske=2025-03-18T02%3A07%3A33Z&skoid=ca7593d4-ee42-46cd-af88-8b886a2f84eb&sks=b&skt=2025-03-17T14%3A07%3A33Z&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skv=2025-01-05&sp=r&spr=https&sr=b&st=2025-03-17T14%3A48%3A43Z&sv=2025-01-05), you can search "CPython 3.10 Win" to find the 3.10 job starting

Reply:
Looking at your logs from your test (https://github.com/henryiii/xgrammar/pull/1) - interesting that it seems to work without issue.

Reply:
I figured out the issue, and it seems kind of strange that this would affect it, but I needed to add the `Interpreter` component. Once I added that things went smoothly. (test [here](https://github.com/Ahajha/xgrammar/actions/runs/13914659118).

Thanks for looking, that gave a good starting point of where else to check!

Reply:
That makes sense. We also included the `Interpreter` prior to migrating to `set(PYBIND11_NEWPYTHON ON)` because the latter seems more convenient.

Thank you both for digging here, and I'm glad this is finally resolved!

Reply:
Yes, you need Interpreter for it to ask the Interpreter where its libs are. Not great for cross-compiling, though, so I'll see if I can get scikit-build-core to report the correct lib location in this case.

Reply:
@henryiii Yeah, that seems rather important. Glad to hear if you have any updates on this!